lib/helpers.nix

```
@@ -140,33 +161,34 @@ with lib; rec {
     # documentation
     mkNullableWithRaw = type: mkNullable (maybeRaw type);
 
-    mkLuaFn = default: desc:
-      lib.mkOption {
-        type = lib.types.nullOr lib.types.str;
-        default = null;
-        description = let
-          defaultDesc = ''
-            default:
-            ```lua
-            ${default}
-            ```
-          '';
-        in
-          if desc == ""
-          then ''
-            (lua function)
+    mkLua = default: desc: let
+      defaultDesc = "default: `${default}`";
+    in
+      mkNullOrLua
+      (
+        if desc == ""
+        then defaultDesc
+        else ''
+          ${desc}
```

```suggestion
    mkLua = default: desc: let
      defaultDesc = "default: `${default}`";
    in
      mkNullOrLua
      (
        (optionalString (desc != "") ''
          ${desc}
          
        '')
        + ''
          default: `${default}`
        '';
```

This looks more readable to me. What do you think ?

---

No real preference